### PR TITLE
crimson/.../scrub_events: fix progress lifetime in deep_scan_object

### PR DESCRIPTION
**Pull request by athanatos**
seastar::repeat may or may not move the passed action after the first invocation, so we can't actually rely on references to variables captured by value being stable.  Instead, allocate with std::make_unique and move into a finally lambda.

Fixes: https://tracker.ceph.com/issues/64996

https://pulpito.ceph.com/sjust-2024-03-21_02:35:36-crimson-rados-wip-sjust-crimson-testing-2024-03-20-distro-default-smithi/

Failures unrelated, caused by existing known bugs.

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by cyx1231st — APPROVED** [review]

**Comment by cyx1231st** [comment]
jenkins test make check

File: src/crimson/osd/osd_operations/scrub_events.cc [hunk 1]
```diff
@@ -258,9 +258,10 @@ ScrubScan::ifut<> ScrubScan::deep_scan_object(
   DEBUGDPP("obj: {}", pg, obj);
   using crimson::common::local_conf;
   auto &entry = ret.objects[obj.hobj];
+  auto progress_ref = std::make_unique<obj_scrub_progress_t>();
+  auto &progress = *progress_ref;
   return interruptor::repeat(
-    [FNAME, this, progress = obj_scrub_progress_t(),
-     &obj, &entry, &pg]() mutable
+    [FNAME, this, &progress, &obj, &entry, &pg]()
```
Comment (tchaikov) [inline src/crimson/osd/osd_operations/scrub_events.cc:264]: as a follow-up, is it possible to transform the `interruptor::repeat()` block to a plain loop in C++, now that the coroutine supported is added to crimson?

**Review by rzarzynski — APPROVED** [review]

File: src/crimson/osd/osd_operations/scrub_events.cc (hunk 1 above)
Comment (athanatos) [reply src/crimson/osd/osd_operations/scrub_events.cc:264]: The next time I make a large enough change here, the whole thing will be a coroutine with normal loops.

**Comment by athanatos** [comment]
jenkins test api

**Review by liu-chunmei — APPROVED** [review]

**Pull request merged**